ls1intum/Artemis
# Create Kubernetes deployment resources

### Motivation and Context


It is planned to migrate Artemis to Kubernetes soon. This PR introduces the initial Kubernetes deployment files including the Artemis application's deployment, its database and the JHipster registry.
It is also planned to migrate Aremis to microservices. Kubernetes deployments are also added for some of the technologies that will be later needed, in this case, Kafka and Gateway. 

### Description

- The Artemis Dockerfile is updated to build a war file that is deployed in Kubernetes.
- Some of the Docker compose files are updated
- Kubernetes resource files are added, including a kustomization file which will deploy all needed applications

### Steps for Testing

1. Use the following documentations to setup a local cluster and deploy Artemis [Kubernetes setup](https://github.com/ls1intum/Artemis/blob/chore/create-kubernetes-files/docs/dev/setup/kubernetes.rst) 
2. Check if Artemis is successfully deployed and running
